### PR TITLE
Move FFI out of core

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/java/org/truffleruby/extra/ffi/PointerNodes.java [hunk 1]
```diff
@@ -70,6 +70,54 @@ public DynamicObject allocate(DynamicObject pointerClass) {
 
     }
 
+    @Primitive(name = "pointer_find_type_size", needsSelf = false)
+    public static abstract class PointerFindTypeSizePrimitiveNode extends PrimitiveArrayArgumentsNode {
+
+        @TruffleBoundary
+        @Specialization(guards = "isRubySymbol(type)")
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/extra/ffi/PointerNodes.java]: I don't know how frequently this is called. Would it make sense to cache the lookup?

**Comment by nirvdrum** [comment]
One side effect of this is the FFI code won't be compiled into the native image, as we currently only compile in core/.

File: src/main/java/org/truffleruby/extra/ffi/PointerNodes.java (hunk 1 above)
Comment (eregon) [reply src/main/java/org/truffleruby/extra/ffi/PointerNodes.java]: It's called for e.g.: `MemoryPointer.new(:int, 4)`. I don't think it's performance critical. We'd need to split the `.new` to make the cache effective.

**Comment by eregon** [comment]
> One side effect of this is the FFI code won't be compiled into the native image, as we currently only compile in core/.

Right, although "compiled" just means pre-parsed here.
Since those files are no longer required during startup it's a gain on startup time there too.

**Pull request merged**